docs/developer-guide.md

```diff
@@ -0,0 +1,66 @@
+# Developer Getting Started Guide
+
+## Pre-requisites
+
+Start w/the following setup guide:
+
+[Setup-Guide-for-Kubernetes-Developers](https://developer.ibm.com/articles/setup-guide-for-kubernetes-developers/)
+
+**Skip the `kubernetes` steps from the guide above if you don't plan on doing any development in the `kubernetes` or `crictl` projects.**
+
+---
+`cri-tools` need the following programs installed in a system in order to work correctly:
+
+- [Go](https://golang.org/doc/install)
+- [Docker](https://docs.docker.com/get-docker/)
+- [Build tools](https://github.com/containerd/cri#install-dependencies)
+
+_**Important Note**: `Go` modules follow a similar structure to projects in Github, i.e. when installing and trying to use `cri-tools` as a local module, it should be installed under a folder structure as follows `go/src/github.com/kubernetes-sigs/cri-tools`. `k8s.io` is the only exception which does not follow this structure for legacy reasons._
+
+After following the setup guides from above, ensure the following:
+
+- There is a `go` directory is within your `$HOME` directory and the `go` binary is part of your environments `$PATH`
+
+  ```bash
+    $ ls $HOME
+    go
+
+    $ echo $PATH | grep go
+    PATH=$PATH:/usr/local/go/bin:$HOME/go/bin # $PATH here refers to truncated version of additional `env` paths that are unrelated this guide / setup
+  ```
+
+## Build & install `containerd/cri`
+
+If you followed the guide so far you may likely already have `containerd` installed, but we will need to install the latest development version.
```

Comment:
If you followed the guide so far you may likely already have `containerd` installed, but we will need to install the latest development version. --> You will require the latest development version of `containerd`.